node/rpcstack.go

```diff
@@ -144,13 +144,15 @@ func (h *httpServer) start() error {
 	h.listener = listener
 	go h.server.Serve(listener)
 
-	// if server is websocket only, return after logging
-	if h.wsAllowed() && !h.rpcAllowed() {
+	if h.wsAllowed() {
 		url := fmt.Sprintf("ws://%v", listener.Addr())
 		if h.wsConfig.prefix != "" {
 			url += h.wsConfig.prefix
 		}
 		h.log.Info("WebSocket enabled", "url", url)
+	}
+	// if server is websocket only, return after logging
+	if !h.rpcAllowed() {
```

Comment:
Hm. Is there ever a case where neither `ws` nor `rpc` is allowed, but we still want to not exit here? I _think_ the answer is no, but .... graphql..? cc @renaynay

Reply:
Graphql cannot be allowed if rpc is disabled